File: src/MyServiceBus.Abstractions/RequestTimeout.cs
namespace MyServiceBus;

public struct RequestTimeout
{
#pragma warning disable THROW001 // Unhandled exception
    static TimeSpan defaultTimeout = TimeSpan.FromSeconds(30);
#pragma warning restore THROW001 // Unhandled exception

    public RequestTimeout(TimeSpan timeSpan)
    {
        this.TimeSpan = timeSpan;
    }

    public TimeSpan TimeSpan { get; }

    public static RequestTimeout None { get; } = new RequestTimeout(TimeSpan.Zero);

    public static RequestTimeout Default { get; } = new RequestTimeout(defaultTimeout);

    public static RequestTimeout After(TimeSpan timeSpan) => new RequestTimeout(timeSpan);
}

File: src/Java/myservicebus-abstractions/src/main/java/com/myservicebus/RequestTimeout.java
package com.myservicebus;

import java.time.Duration;

public final class RequestTimeout {
    private static final Duration DEFAULT_DURATION = Duration.ofSeconds(30);

    private final Duration duration;

    public RequestTimeout(Duration duration) {
        this.duration = duration;
    }

    public Duration getDuration() {
        return duration;
    }

    public static final RequestTimeout NONE = new RequestTimeout(Duration.ZERO);
    public static final RequestTimeout DEFAULT = new RequestTimeout(DEFAULT_DURATION);

    public static RequestTimeout after(Duration duration) {
        return new RequestTimeout(duration);
    }
}
